Modules/ArcGIS/DSCResources/ArcGIS_Federation/ArcGIS_Federation.psm1

```diff
@@ -127,8 +127,20 @@ function Test-TargetResource
 	[System.Reflection.Assembly]::LoadWithPartialName("System.Web") | Out-Null	    
     Write-Verbose "Get Portal Token from Deployment '$PortalHostName'"
     $Referer = "https://$($PortalHostName)/$PortalContext"
-    $token = Get-PortalToken -PortalHostName $PortalHostName -Port $PortalPort -SiteName $PortalContext `
-         -Credential $RemoteSiteAdministrator -Referer $Referer 
+    $waitForToken = $true
+    $waitForTokenCounter = 0 
+    while ($waitForToken -and $waitForTokenCounter -lt 5) {
+        $waitForTokenCounter++
+        try{
+            $token = Get-PortalToken -PortalHostName $PortalHostName -Port $PortalPort -SiteName $PortalContext -Credential $RemoteSiteAdministrator -Referer $Referer
+        } catch {
+            Write-Verbose "Error getting Token for Federation ! Waiting for 5 Minutes to try again"
+            Start-Sleep -Seconds 300
```

Comment:
5 minute sleep might be excessive. Perhaps decrease sleep time and increase the number of retries, so that we don't "waste" time